src/scripts/main.js

```diff
@@ -1 +1,24 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const mainImage = document.querySelector('#largeImg');
+
+  const thumbnails = document.querySelector('#thumbs');
+
+  function changeImage(e) {
+    const necessaryLink = e.target.closest('a');
+
+    if (necessaryLink) {
+      e.preventDefault();
+
+      const clickedImage = necessaryLink.querySelector('img');
+
+      if (clickedImage) {
+        const newSrc = necessaryLink.getAttribute('href');
+
+        mainImage.setAttribute('src', newSrc);
+      }
+    }
+  }
```

Comment:
The function changeImage is doing more than one thing. It's generally a good practice to have functions do one thing only. In this case, the function is not only changing the image but also checking if the clicked element is a link and if it has an image. Consider breaking it down into smaller, more specific functions.